version: '3'
services:
  postgres-db:
    image: postgres:14.1-alpine
    volumes:
      - ./tmp/db:/var/lib/postgresql/data
    ports:
      - '5432:5432'
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=PgAdmin2024!
      - POSTGRES_DB=postgresdb-seupc-dev
    networks:
      - application
    container_name: postgres-db
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U postgres']
      interval: 5s
      timeout: 5s
      retries: 5

  core-api:
    build:
      context: ./apps/core-api/.
      dockerfile: ./Dockerfile
    image: core-api:v1
    ports:
      - '8080:8080'
    volumes:
      - './apps/core-api:/apps/core-api/'
    networks:
      - application
    container_name: core-api
    depends_on:
      - postgres-db
    environment:
      - DATABASE_URL=postgres://postgres:PgAdmin2024!@postgres-db:5432/postgresdb-seupc-dev

  client-app:
    build:
      context: ./apps/client-app/.
      dockerfile: ./Dockerfile
    image: client-app
    ports:
      - '3000:3000'
    volumes:
      - './apps/client-app/:/apps/client-app/'
    networks:
      - application
    container_name: client-app
    depends_on:
      - core-api

networks:
  application:
    driver: bridge
